jupyterhub/app.py

```diff
@@ -544,6 +544,20 @@ def _authenticator_default(self):
         """
     ).tag(config=True)
 
+    concurrent_spawn_limit = Integer(
+        None,
+        allow_none=True,
+        help="""
+        Maximum number of concurrent users that can be spawning at a time.
+
+        If more than this many users attempt to spawn at a time, they
+        enter an exponential delay with a timeout.
```

Comment:
Exponential delay isn't implemented. When the limit is reached, it fails with 429 and it is incumbent on the user to retry later. I think the current behavior is sensible, but we should update the message to be accurate.